apps/cowswap-frontend/src/legacy/utils/trade.ts

```diff
@@ -71,7 +87,8 @@ export function getOrderSubmitSummary(
   ]
   const inputSymbol = formatSymbol(sellToken.symbol)
   const outputSymbol = formatSymbol(buyToken.symbol)
-  const inputAmountValue = formatTokenAmount(feeAmount ? inputAmount.add(feeAmount) : inputAmount)
+  // this already contains the fee in the fee amount when fee=0
+  const inputAmountValue = formatTokenAmount(feeAmount && !swapZeroFee ? inputAmount.add(feeAmount) : inputAmount)
```

Comment:
The values used here are based on what's signed.
- fee=0 on => fee already included in the sell amount
- fee=0 off => fee NOT included in the sell amount